scripts/irods/test/test_izonereport.py

```diff
@@ -86,14 +94,78 @@ def test_izonereport_and_validate(self):
-
-            self.assertIn('resources', icat_server_object.keys())
-            self.assertGreaterEqual(len(icat_server_object['resources']), 1)
-            for resource in icat_server_object['resources']:
+            
+            zone_info = json.loads(stdout)['zones'][0]
+            server_array = zone_info['servers']
+            catalog_server = server_array[0]
+            for server in server_array:
+                if server['server_config']['catalog_service_role'] == 'provider':
+                    catalog_server = server
+                    break
+                
+            self.assertIn('resources', catalog_server.keys())
+            self.assertGreaterEqual(len(catalog_server['resources']), 1)
+            for resource in catalog_server['resources']:
                 self.assertIn('id', resource.keys())
 
-            self.assertIn('coordinating_resources', icat_server_object.keys())
-            self.assertGreaterEqual(len(icat_server_object['coordinating_resources']), 1)
-            for resource in icat_server_object['coordinating_resources']:
+            self.assertIn('coordinating_resources', zone_info.keys())
+            self.assertGreaterEqual(len(zone_info['coordinating_resources']), 1)
+            for resource in zone_info['coordinating_resources']:
                 self.assertIn('id', resource.keys())
+
+    # see issue #6857
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python' or test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing or python rule engine")
+    def test_servers_are_flattened_core(self):
+        
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        self.assertIn('servers', zone_info.keys())
+        self.assertNotIn('catalog_server', zone_info.keys())
+        self.assertEqual(len(zone_info['servers']), 1)
+        
+        self.assertIn('coordinating_resources', zone_info.keys())
+        self.assertEqual(len(zone_info['coordinating_resources']), 3)
+        
+    # see issue #6857
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python' or not test.settings.RUN_IN_TOPOLOGY, "Skip for Core Testing or python rule engine")
+    def test_servers_are_flattened_topology(self):
+        
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        self.assertIn('servers', zone_info.keys())
+        self.assertNotIn('catalog_server', zone_info.keys())
+        self.assertEqual(len(zone_info['servers']), 4)
+        
+        self.assertIn('coordinating_resources', zone_info.keys())
+        self.assertEqual(len(zone_info['coordinating_resources']), 3)
+        
+    # see issue #6856
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python', "only run for native rule language")
+    def test_plugin_calculate_checksums(self):
```

Comment:
Consider using more words to express what the test is about.

For example:
```python
def test_zone_report_calculates_checksums_for_all_plugins__issue_6856(self):
```

Again, this is simply an example.